src/sema/function_annotation.rs

```diff
@@ -0,0 +1,184 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{
+    ast::{ConstructorAnnotation, Diagnostic, Namespace, Type},
+    diagnostics::Diagnostics,
+    expression::{expression, ExprContext, ResolveTo},
+    unused_variable::used_variable,
+    Symtable,
+};
+use crate::Target;
+use solang_parser::pt::{self, CodeLocation};
+
+/// Collect the seeds, bump, payer, and space for constructors. This is a no-op on Substrate/EVM since
+/// there should be no seed or bump annotations permitted on other targets.
+pub fn function_annotations(
+    function_no: usize,
+    annotations: &[&pt::Annotation],
+    symtable: &mut Symtable,
+    context: &ExprContext,
+    ns: &mut Namespace,
+) {
+    let mut diagnostics = Diagnostics::default();
+
+    // On Solana, the seeds and bump for a constructor can be specified using annotations, for example
+    //
+    // @seed(param1)
+    // @seed("fizbaz")
+    // @bump(param2)
+    // constructor(bytes param1, uint8 param2) {}
+
+    let mut parsed_annotations = Vec::new();
+    let mut bump = None;
+    let mut space = None;
+    let mut payer = None;
+
+    for note in annotations {
+        if ns.target == Target::Solana
```

Comment:
The target won't change inside the loop, so you better check it outside. The same applies for the function type.

Reply:
I've created a temporary for this.

Having said this,iterating over annotations does not need optimizing, if anything it only needs optimizing for code readability, not runtime execution speed.

Reply:
Why do you keep checking this at every branch? I don't understand why you can't just check the condition before the for-loop or even return early in function.